source/patterns/@aws-solutions-constructs/aws-wafwebacl-apigateway/test/test.wafwebacl-apigateway.test.ts

```diff
@@ -0,0 +1,361 @@
+      {
+        Name: "AWS-AWSManagedRulesAnonymousIpList",
+        OverrideAction: {
+          None: {}
+        },
+        Priority: 3,
+        Statement: {
+          ManagedRuleGroupStatement: {
+            Name: "AWSManagedRulesAnonymousIpList",
+            VendorName: "AWS"
+          }
+        },
+        VisibilityConfig: {
+          CloudWatchMetricsEnabled: true,
+          MetricName: "AWSManagedRulesAnonymousIpList",
+          SampledRequestsEnabled: true
+        }
+      },
+      {
+        Name: "AWS-AWSManagedRulesAmazonIpReputationList",
+        OverrideAction: {
+          None: {}
+        },
+        Priority: 4,
+        Statement: {
+          ManagedRuleGroupStatement: {
+            Name: "AWSManagedRulesAmazonIpReputationList",
+            VendorName: "AWS"
+          }
+        },
+        VisibilityConfig: {
+          CloudWatchMetricsEnabled: true,
+          MetricName: "AWSManagedRulesAmazonIpReputationList",
+          SampledRequestsEnabled: true
+        }
+      },
+      {
+        Name: "AWS-AWSManagedRulesAdminProtectionRuleSet",
+        OverrideAction: {
+          None: {}
+        },
+        Priority: 5,
+        Statement: {
+          ManagedRuleGroupStatement: {
+            Name: "AWSManagedRulesAdminProtectionRuleSet",
+            VendorName: "AWS"
+          }
+        },
+        VisibilityConfig: {
+          CloudWatchMetricsEnabled: true,
+          MetricName: "AWSManagedRulesAdminProtectionRuleSet",
+          SampledRequestsEnabled: true
+        }
+      },
+      {
+        Name: "AWS-AWSManagedRulesSQLiRuleSet",
+        OverrideAction: {
+          None: {}
+        },
+        Priority: 6,
+        Statement: {
+          ManagedRuleGroupStatement: {
+            Name: "AWSManagedRulesSQLiRuleSet",
+            VendorName: "AWS"
+          }
+        },
+        VisibilityConfig: {
+          CloudWatchMetricsEnabled: true,
+          MetricName: "AWSManagedRulesSQLiRuleSet",
+          SampledRequestsEnabled: true
+        }
+      }
+    ]
+  });
+});
+
+// --------------------------------------------------------------
+// Test web acl with user provided acl props
+// --------------------------------------------------------------
+test('Test user provided acl props', () => {
+  const stack = new cdk.Stack();
+  const inProps: lambda.FunctionProps = {
+    code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    runtime: lambda.Runtime.NODEJS_10_X,
+    handler: 'index.handler'
+  };
+  const func = defaults.deployLambdaFunction(stack, inProps);
+  const [_api] = defaults.RegionalLambdaRestApi(stack, func);
+  const customRules = [
+    wrapManagedRuleSet("AWSManagedRulesCommonRuleSet", "AWS", 0),
+    wrapManagedRuleSet("AWSManagedRulesWordPressRuleSet", "AWS", 1),
+  ];
+
+  const wafAcl = new waf.CfnWebACL(stack, 'test-waf', {
+    defaultAction: {
+      allow: {}
+    },
+    scope: 'REGIONAL',
+    visibilityConfig: {
+      cloudWatchMetricsEnabled: true,
```

Comment:
This changes only the rules, the rest of the values appear to match the defaults. Since it looks like the actual construct code does not override any of the props but the rules, this input data set is the only data set that would pass this test. If you set cloudWatchMetricsEnabled to false, I believe this test would fail.

This comment really applies to several tests in this module that have results that are dozens of lines long.